app/controllers/admissions_admin/groups_controller.rb

```diff
@@ -51,4 +51,58 @@ def reject_calls
     @group = Group.find(params[:id])
     @applicants_to_call = @group.applicants_to_call(@admission).sort_by(&:full_name)
   end
+
+  def show_unlogged_applicants
+    @admission = Admission.find(params[:admission_id])
+    @group = Group.find(params[:group_id])
+
+    @group_applicants = {}
+    @group_applicants[@group] = []
+    @group.jobs.where(admission: @admission).each do |job|
+      applicants_in_job = job.job_applications.map(&:applicant).flatten
+      unlogged_in_job = applicants_in_job.select do |a|
+        a.log_entries.empty?
+      end
+      job_and_applicants = JobAndApplicants.new(job, unlogged_in_job)
+      @group_applicants[@group].push(job_and_applicants)
+    end
+
+    @group_jobs = @group.jobs.where(admission: @admission).sort
+    @all_applicants = @group_jobs
+                        .map(&:job_applications)
+                        .flatten.map(&:applicant)
+                        .uniq
+
+    @unlogged_applicants = @group.unlogged_applicants(@admission)
+    @log_entries = LogEntry.possible_log_entries
+
+    @submit_url = admissions_admin_admission_group_show_unlogged_applicants_path(@admission, @group, format: :json)
+    @back_url = admissions_admin_admission_group_path(@admission, @group)
+
+    @email_text = @unlogged_applicants.map(&:email).join('\n')
+  end
+
+  def log_all_applicants
+    admission = Admission.find(params[:admission_id])
+    text = params[:log_entry][:log]
+    admission.log_applicants_in_group(@group.unlogged_applicants, text, @group, @current_user)
+
+    flash[:success] = t('admissions_admin.all_applicants_logged')
+    redirect_to admissions_admin_admission_group_show_unlogged_applicants_path
+  end
+
+  def log_single_applicant
+    admission = Admission.find(params[:admission_id])
+    applicant = Applicant.find(params[:applicant_id])
+    group = Group.find(params[:group_id])
+    text = params[:log_entry][:log]
+
+    applicant.log_with_text(text, group, admission, @current_user)
```

Comment:
If you want to validate that this works you can implement a flow like
```
if applicant.log_with_text(text, group, admission, @current_user)
  flash[:success] = ...
else
  flash[:error] = something went wrong
end
```

as `save` in `log_with_text` returns `false` if anything fails
https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-save
This is for Rails 6 but I imagine that it hasnt changed from Rails 5.

Not that this matters that much, if any at all, because this is very exclusive internal functionality

Reply:
Ah, yes, this is very nice :) Will implement!